### PR TITLE
Yields the component so the block form is actually useful

### PR DESCRIPTION
**Pull request by poteto**

### CODE REVIEW

**Comment by bcardarella** [comment]
@danmcclain how did this break? I thought `hasBlock` was already a CP on components.

Even more strange, we have a test for this https://github.com/dockyard/ember-cli-async-button/blob/master/tests/acceptance/button-test.js#L191-L197 and that was green.

**Comment by danmcclain** [comment]
Unsure how the test was even green! Thought hasBlock was part of component now, but tests should have proven otherwise. Will have to look into making sure test fails properly.

**Comment by danmcclain** [comment]
Ah, hasBlock was provided as a polyfill, per @poteto's commit comments. Ultimately it is unneeded, as this version won't work with pre 1.13 ember due to the positional params 

Yielding the component is a feature I missed, :+1:

**Comment by rwjblue** [comment]
Can we have a test that exercises the yielded block param, to prevent future regressions?

**Comment by danmcclain** [comment]
@rwjblue Good idea

Also, the docs for hasBlock, showing that we aren't insane: http://emberjs.com/api/classes/Ember.Component.html#property_hasBlock :smile:

**Comment by rwjblue** [comment]
@danmcclain - `hasBlock` is absolutely available in templates, as those docs show.  It is not available inside the component JS though.

File: addon/components/async-button.js [hunk 1]
```diff
@@ -4,8 +4,10 @@ import layout from '../templates/components/async-button';
 var get = Ember.get;
 var getWithDefault = Ember.getWithDefault;
 var set = Ember.set;
+var computed = Ember.computed;
 
 export default Ember.Component.extend({
+  hasBlock: computed.bool('template').readOnly(),
```
Comment (rwjblue) [inline addon/components/async-button.js]: Accessing `template` in 1.13 will cause a deprecation.  This polyfill does work on 1.12, but `positionalParams` [used here](https://github.com/dockyard/ember-cli-async-button/blob/master/addon/components/async-button.js#L9) will not work at all on versions prior to 1.13.

If this is useful even without `positionalParams` (maybe the params aren't needed in all cases?), then we should only add the polyfill when we are in Ember versions prior to 1.13.

If this is not useful without `positionalParams` then we shouldn't add the polyfill at all, and we should warn when using with an older Ember version.  Something like [this in ember-get-helpers](https://github.com/jmurphyau/ember-get-helper/blob/master/index.js#L18) or [this in ember-auto-register-helpers](https://github.com/stefanpenner/ember-cli-auto-register-helpers/blob/master/index.js#L18).

**Comment by bcardarella** [comment]
If the goal of this PR is to support versions prior to Ember 1.13 then I don't want to do that. Supporting the multiple versions between 1.10 - 2.x is a nightmare and I'd much rather say "you're using 1.10, go use this version"

**Comment by poteto** [comment]
The goal was to yield `this` properly, I'm adding a test for it

**Comment by danmcclain** [comment]
Wasn't broken before, just adding the yeilding of this :wink:

**Comment by bcardarella** [comment]
@poteto can we remove the polyfill if that isn't necessary for 1.13

**Comment by poteto** [comment]
Done!

**Comment by danmcclain** [comment]
:+1:

**Comment by bcardarella** [comment]
And finally, an update to the CHANGELOG: https://github.com/dockyard/ember-cli-async-button/blob/master/CHANGELOG.md

**Comment by bcardarella** [comment]
thanks!

**Pull request merged**